daemon/start.go

```diff
@@ -31,12 +31,22 @@ func (daemon *Daemon) ContainerStart(name string, hostConfig *containertypes.Hos
 		// creating a container, not during start.
 		if hostConfig != nil {
 			logrus.Warn("DEPRECATED: Setting host configuration options when the container starts is deprecated and will be removed in Docker 1.12")
+			oldNetworkMode := container.HostConfig.NetworkMode
 			if err := daemon.setSecurityOptions(container, hostConfig); err != nil {
 				return err
 			}
 			if err := daemon.setHostConfig(container, hostConfig); err != nil {
 				return err
 			}
+			newNetworkMode := container.HostConfig.NetworkMode
+			if string(oldNetworkMode) != string(newNetworkMode) {
```

Comment:
@mavenugo  `newNetworkMode` couldn't be empty, since the `hostConfig` that `ContainerStart` passed is from `runconfig.DecodeHostConfig` (https://github.com/docker/docker/blob/master/api/server/router/container/container_routes.go#L174 ) and `DecodeHostConfig` will also call `SetDefaultNetModeIfBlank`(called in `getHostConfig`)